tests/coap_nanocoap/README.md

```diff
@@ -0,0 +1,18 @@
+# nanocoap Test App
+Provides a CLI based tool for interactive or scripted testing of nanocoap client
+and server operation. Top level commands:
+
+## *server*
+Options are limited to *start* to start listening. Provides these resources:
+
+`/value`<br>
+read/write an unsigned 8-bit integer
+
+`/.well-known/core`<br>
+Read the list of resources. Expects a block2 based request or else returns the
+first 16 bytes of the list.
+
+## *client*
+Provides get/put/post to an address, with an optional payload. Enter the command
+with no other text (just *client*) to see command syntax. Options are similar to
```

Comment:
Removed reference to gcoap example in e90e665. 

Must be confirmable because that is a requirement of the `nanocoap_request()` function. It should be a simple change to allow for non-confirmable requests. I will add this task as a follow-up to my upcoming PR for #9191 because that issue fixes some of the handling around retries. After that, we can add a parameter to this test to specify confirmable or non-confirmable.